src/trustchain.rs

```diff
@@ -122,3 +126,239 @@ impl TrustchainContext {
         ))
     }
 }
+
+#[pyclass]
+#[derive(PartialEq)]
+pub(crate) struct TrustchainError(pub(crate) libparsec::core::TrustchainError);
+
+// This object is only here to wrap our `TrustchainError` in a python exception
+// object. Without this object we can't raise or except `TrustchainError` in
+// python code
+create_exception!(_parsec, TrustchainErrorException, PyException);
+
+#[pymethods]
+impl TrustchainError {
```

Comment:
Do we really need all of those constructor ? For me the exception was not created in the python code.

> Since that exception is not created in the python code base you don't need to provide a way for python to create it